src/extension/infer.rs

```diff
@@ -1091,4 +1091,63 @@ mod test {
         }
         Ok(())
     }
+
+    #[test]
+    fn extension_adding_sequence() -> Result<(), Box<dyn Error>> {
+        fn mknode(
+            hugr: &mut Hugr,
+            root: Node,
+            sig: FunctionType,
+            ext: ExtensionId,
+        ) -> Result<Node, Box<dyn Error>> {
+            let [node, input, output] = create_with_io(
+                hugr,
+                root,
+                ops::DFG {
+                    signature: sig
+                        .clone()
+                        .with_extension_delta(&ExtensionSet::singleton(&ext)),
+                },
+            )?;
+
+            let lift = hugr.add_node_with_parent(
+                node,
+                NodeType::open_extensions(ops::LeafOp::Lift {
+                    type_row: type_row![NAT],
+                    new_extension: ext,
+                }),
+            )?;
+
+            hugr.connect(input, 0, lift, 0)?;
+            hugr.connect(lift, 0, output, 0)?;
+
+            Ok(node)
+        }
+
+        let df_sig = FunctionType::new(type_row![NAT], type_row![NAT]);
+
+        let mut hugr = Hugr::new(NodeType::open_extensions(ops::DFG {
+            signature: df_sig
+                .clone()
+                .with_extension_delta(&ExtensionSet::from_iter([A, B])),
+        }));
+
+        let root = hugr.root();
+
+        let node0 = mknode(&mut hugr, root, df_sig.clone(), A)?;
+        let node1 = mknode(&mut hugr, root, df_sig.clone(), A)?;
+        let node2 = mknode(&mut hugr, root, df_sig.clone(), B)?;
+        let node3 = mknode(&mut hugr, root, df_sig.clone(), B)?;
+        let node4 = mknode(&mut hugr, root, df_sig.clone(), A)?;
+        let node5 = mknode(&mut hugr, root, df_sig.clone(), B)?;
+
+        // Connect nodes in order (0 -> 1 -> 2 ...)
+        let nodes = [node0, node1, node2, node3, node4, node5];
+        let mut windows = nodes.windows(2);
+        while let Some(&[src, tgt]) = windows.next() {
```

Comment:
Can this be `for &[src, tgt] in nodes.windows(2) {...}` or something like that?

Reply:
Unfortunately that doesn't work quite as well because rust doesn't know at compile time the size of the window and complains that `&[src, tgt]` is a refutable pattern.

Reply:
You can use [`tuple_windows](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.tuple_windows) instead.

Reply:
🙌